components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -762,13 +773,27 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     ): Promise<boolean> {
         const phoneNumber = formatPhoneNumber(rawPhoneNumber);
         const user = this.checkUser("verifyPhoneNumberVerificationToken");
-        const checked = await this.verificationService.verifyVerificationToken(phoneNumber, token);
-        if (!checked) {
+        const { verified, channel } = await this.verificationService.verifyVerificationToken(phoneNumber, token);
+        if (!verified) {
+            this.analytics.track({
+                event: "phone_verification_failed",
+                userId: user.id,
+                properties: {
+                    channel,
+                },
+            });
```

Comment:
As a follow-up, it may be useful to generate a Verification ID on the server side, and give it to the client and send it back on the validation phase, such that the same ID can be used in both events - this would allow us to join on the same verification flow, and assert how many cases there are where the verification doesn't complete - useful for abuse tracking anyway